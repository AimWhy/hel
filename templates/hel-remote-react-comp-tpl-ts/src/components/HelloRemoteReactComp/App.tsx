import React from 'react';
import './App.css';
import logo from './logo.svg';

interface IProps {
  onHeaderClick?: (label: string) => void;
}

const App = React.forwardRef(function App(props: IProps, ref: any) {
  console.log('hi ref', ref);

  React.useImperativeHandle(ref, () => ({
    sayHello() {
      alert('Hello hel-micro remote comp (ts)');
    },
  }));
  return (
    <div className="App">
      <header className="App-header" onClick={() => props.onHeaderClick?.('header click')}>
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          <h1>This is a hel remote react component (ts) v2</h1>
          <h2>emitted by hel-micro</h2>
        </p>
        <a className="App-link" href="https://tnfe.github.io/hel" target="_blank" rel="noopener noreferrer">
          Learn hel-micro
        </a>
      </header>
    </div>
  );
});

export default App;
