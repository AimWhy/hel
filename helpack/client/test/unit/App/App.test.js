import App from 'App';
import { shallow } from 'enzyme';

jest.mock('utils/http');
/**
 * @author billyjwang
 * @priority P0
 * @casetype unit
 */
describe('App 组件单元测试', () => {
  test('生成 App 组件快照', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
