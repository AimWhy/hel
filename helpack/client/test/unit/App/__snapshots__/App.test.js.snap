// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`{"name":"App 组件单元测试","author":"billyjwang","priority":"P0","casetype":"unit"} {"name":"生成 App 组件快照","author":"billyjwang","priority":"P0","casetype":"unit"} 1`] = `<OmApp />`;
